plasma_framework/contracts/src/exits/payment/controllers/PaymentChallengeIFENotCanonical.sol

```diff
@@ -0,0 +1,125 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import "../PaymentExitDataModel.sol";
+import "../routers/PaymentInFlightExitRouterArgs.sol";
+import "../spendingConditions/IPaymentSpendingCondition.sol";
+import "../spendingConditions/PaymentSpendingConditionRegistry.sol";
+import "../../utils/ExitId.sol";
+import "../../../utils/UtxoPosLib.sol";
+import "../../../utils/Merkle.sol";
+import "../../../framework/PlasmaFramework.sol";
+
+library PaymentChallengeIFENotCanonical {
+    using UtxoPosLib for UtxoPosLib.UtxoPos;
+
+    struct Controller {
+        PlasmaFramework framework;
+        PaymentSpendingConditionRegistry spendingConditionRegistry;
+    }
+
+    event InFlightExitChallenged(
+        address indexed challenger,
+        bytes32 txHash,
+        uint256 challengeTxPosition
+    );
+
+    function run(
+        Controller memory self,
+        PaymentExitDataModel.InFlightExitMap storage inFlightExitMap,
+        PaymentInFlightExitRouterArgs.ChallengeCanonicityArgs memory args
+    )
+        public
+    {
+        // Check if there is an active in-flight exit from this transaction?
```

Comment:
Minor: After solidity 0.5, `require()` can be set with a message. I feel some previous comments become trivial after that.
eg. `require(ife.exitStartTimestamp != 0, "In-fligh exit doesn't exists");` pretty much explains itself.